windows/src/engine/keyman32/kmprocessactions.cpp

```diff
@@ -6,94 +6,71 @@
  */
 #include "pch.h"
 
-static BOOL processUnicodeChar(AITIP* app, const km_core_action_item* actionItem) {
-  if (Uni_IsSMP(actionItem->character)) {
-    app->QueueAction(QIT_CHAR, (Uni_UTF32ToSurrogate1(actionItem->character)));
-    app->QueueAction(QIT_CHAR, (Uni_UTF32ToSurrogate2(actionItem->character)));
-  }
-  else {
-    app->QueueAction(QIT_CHAR, actionItem->character);
+static void
+processUnicodeChar(AITIP* app, const km_core_usv* actionItem) {
```

Comment:
My refactor here has left a confusing variable name behind. `actionItem` should be `ouput_string` as defined in the `action_struct->output //null-term string of characters to insert into the document.

Reply:
> Thus null (U+0000) isn't permitted?

Null has never been permitted in Keyman for Windows. Assumption has been, since very beginning of Keyman, null terminated strings everywhere. .kmx has null terminated strings, so it would be impossible to emit one from a .kmx anyway. So while it is a theoretical change in behaviour, in practice it doesn't matter.

Reply:
ok. kmx+ technically should support null chars (ch=0000, extend=false)

Reply:
I'm just wondering if kmc should error on say `\u{0}`

Reply:
Yes, we should error on that. I don't want our keyboards to emit nulls. In practice most apps will ignore but there's no legit use case that I can see for language keyboards to emit nulls. Perhaps some geek somewhere wants to do that but they can write their own keyboard manager.

Reply:
In practice, most of the control characters should probably be errored away. Allow TAB, CR, LF. That's about it. Everything else is going to cause interop issues.